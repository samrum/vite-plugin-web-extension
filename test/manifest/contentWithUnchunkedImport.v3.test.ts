import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("contentWithUnchunkedImport");

runManifestV3Test("contentWithUnchunkedImport", () => ({
  content_scripts: [
    {
      js: [`${resourceDir}/content.js`],
      matches: [
        "*://*/*",
        "https://*/*",
        "*://example.com/",
        "https://example.com/",
        "*://example.com/subpath/*",
        "https://example.com/subpath/*",
      ],
    },
  ],
}));
