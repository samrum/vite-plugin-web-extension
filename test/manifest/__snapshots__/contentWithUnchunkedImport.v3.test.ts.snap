// Vitest Snapshot v1

exports[`contentWithUnchunkedImport - Manifest V3 1`] = `
[
  {
    "code": "function log(message) {
  console.log(message);
}
log(\\"content\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "/mocked-repo-dir/test/manifest/resources/contentWithUnchunkedImport/content.js",
    "fileName": "assets/test/manifest/resources/contentWithUnchunkedImport/content.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "/mocked-repo-dir/test/manifest/resources/contentWithUnchunkedImport/content.js": {
        "code": "log(\\"content\\");",
        "originalLength": 52,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 15,
      },
      "/mocked-repo-dir/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "test/manifest/resources/contentWithUnchunkedImport/content",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 3,
  \\"content_scripts\\": [
    {
      \\"js\\": [
        \\"assets/test/manifest/resources/contentWithUnchunkedImport/content.js\\"
      ],
      \\"matches\\": [
        \\"*://*/*\\",
        \\"https://*/*\\",
        \\"*://example.com/\\",
        \\"https://example.com/\\",
        \\"*://example.com/subpath/*\\",
        \\"https://example.com/subpath/*\\"
      ]
    }
  ]
}",
    "type": "asset",
  },
]
`;
